hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/KeyManagerImpl.java

```diff
@@ -871,6 +871,19 @@ private boolean isKeyEmpty(OmKeyInfo keyInfo) {
      startKeyName, keyPrefix, maxKeys);
   }
 
+  @Override
+  public boolean recoverTrash(String volumeName, String bucketName,
+      String keyName, String destinationBucket) throws IOException {
+
+    Preconditions.checkNotNull(volumeName);
```

Comment:
Not sure the @Nonnull is a better way? I saw the Nonnull annotation in the  hadoop-ozone project already, and findbugs will find null argument during build.

Reply:
Thanks @maobaolong for this advice.

As I know, we could use both ```@Nonnull``` and ```Preconditions.checkNotNull()```, and the former is to inform developer not use ```null``` in the parameter, and the later is to validate in runtime.

So, I think this part would not be fixed, and we could create a Jira for the annotation issue.

Reply:
ok